### PR TITLE
Adding some documentation on setting up LSP for neovim 0.8+

### PR DESCRIPTION
**Pull request by scottfines**
Should be a minor change to the Editors README on how to set up sourcekit-lsp for nevim 0.8 and above. The code segments are taken from my own tested configuration, along with as much notation as I could be reasonably confident in.

### CODE REVIEW

**Review by etcwilde — APPROVED** [review]
The indentation looks maybe a bit excessive (though personal opinion), the code snippet themselves look fine. Maybe add a few links to documentations on the tools so that folks can configure the server for themselves.

e.g https://neovim.io/doc/user/lsp.html#lsp-config, https://neovim.io/doc/user/lsp.html, etc...

**Comment by scottfines** [comment]
I adjusted the indentation to line up with what I see in the rest of the doc (2 spaces), hopefully that will be more consistent. Also added a link to neovim's LSP configuration page for further information. There are some plugins that (theoretically) smooth this out, but since I haven't used them, I'm somewhat reluctant to even tangentially endorse them in the docs.

File: Editors/README.md [hunk 1]
```diff
@@ -143,6 +143,54 @@ As a test, open a Swift file, put the cursor on top of a symbol in normal mode a
 :call CocAction('jumpDefinition')
 ```
 
+### Neovim 0.8 and above
+since version 0.8, neovim has native LSP support, which can be used to connect to sourcekit-lsp directly. To do so, add the following to 
+a .lua config file (such as `lua/swift.lua`):
+
+```lua
+require'lspconfig'.sourcekit.setup{
+        cmd = {'$TOOLCHAIN_PATH/usr/bin/sourcekit-lsp'}
```
Comment (etcwilde) [inline Editors/README.md:152]: One more to adjust the indentation on. :)

**Review by etcwilde — APPROVED** [review]
Thanks for making those changes.

>  There are some plugins that (theoretically) smooth this out, but since I haven't used them, I'm somewhat reluctant to even tangentially endorse them in the docs.

I've been pretty happy with `vim-lsp`, though that's listed above. It works reasonably well with neovim too. I haven't tried any of the neovim/lua specific versions either.

File: Editors/README.md (hunk 1 above)
Comment (scottfines) [reply Editors/README.md]: oops! my bad! fixed :)

**Comment by etcwilde** [comment]
@swift-ci please smoke test

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
Windows failure:
```
[swift]                                 Updating 'C:\Users\swift-ci\jenkins\workspace\sourcekit-lsp-PR-windows\swift'
[swift]                                 + git fetch --recurse-submodules=yes --tags
[swift]                                 b'From https://github.com/apple/swift'
[swift]                                 b' * branch                    HEAD       -> FETCH_HEAD'
[swift]                                 + git checkout main
[swift]                                 b"error: pathspec 'main' did not match any file(s) known to git"
```

**Comment by etcwilde** [comment]
@swift-ci please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by thecoldstone** [comment]
Just my 5 cents. After testing with different setups the most easiest one and less requiring does work perfectly with Neovim :)

**Pull request merged**